### ISSUE TITLE
performance: current creates new copies of the objects where unnecessary

### ISSUE THREAD

**Issue by francescotescari** [unadlib/mutative#47]
`current` creates a copy of nested objects, even when such objects are not draft.
Consider the following example:

```ts
  const obj = { k: 42 };
  const original = { x: { y: { z: [obj] }}}
  const yReplace = { z: [obj] } ;

  // with create
  const withCreate = create(original, draft => {
    draft.x.y = yReplace;
  });
  console.log(withCreate.x.y === yReplace) // prints true
  console.log(withCreate.x.y.z[0] === obj) // prints true

  // with draft + current
  const [draft] = create(original);
  draft.x.y = yReplace;
  const withDraft = current(draft);
  console.log(withDraft.x.y === yReplace) // prints false
  console.log(withDraft.x.y.z[0] === obj) // prints false! DEEP COPY???
```
I would expect the draft + current to behave like the create option, returning the new object, but currently actually performs a deep copy instead. This has a big negative impact on the performance of `current`

**Comment by unadlib** [comment]
Thank you for submitting this issue and PR.

**Comment by unadlib** [comment]
Perhaps tomorrow I will do some minor refactoring and release a new version.

**Comment by unadlib** [comment]
hi @francescotescari, I have released Mutative v1.0.7. Feel free to use it.
